### PR TITLE
Bug fixes for LLVM Backend

### PR DESCRIPTION
**Pull request by dwightguth**
I'm going to try to release a PR every day with the smaller fixes for the day rather than letting it clump up so much, in the future, but this is what has built up so far. Let me know if there's specific things you'd like to see given regression tests

### CODE REVIEW

**Comment by traiansf** [comment]
I was wondering whether we could have Jenkins run all Haskell backend tests whenever there is a change to ModuleToKore or to the supporting transformations (such as AddSortInjections).  This part is known to be fragile, so making it sure it passes tests would be better done now (since we know we've touch potentially sensitive code) rather than waiting for @ehildenb 's job to do it one in a while.

Other than that, the changes look reasonable to me.

**Comment by traiansf** [comment]
```
 make[1]: *** [or-llvm] Error 2
```

File: kernel/src/main/java/org/kframework/compile/AddSortInjections.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ private K internalAddSortInjections(K term, Sort expectedSort) {
                         }
                     }
                 }
-                throw new AssertionError();
+                return KApply(KLabel("inj", actualSort, expectedSort), KList(visitChildren(term, actualSort)), Att.empty().add(Sort.class, expectedSort));
             } else {
                 return KApply(KLabel("inj", actualSort, expectedSort), KList(visitChildren(term, actualSort)), Att.empty().add(Sort.class, expectedSort));
```
Comment (ehildenb) [inline kernel/src/main/java/org/kframework/compile/AddSortInjections.java]: I think we can drop the `else` branch and hoist these identical `return KApply` to after the `if_else`.

**Review by ehildenb — APPROVED** [review]
Overall looks pretty good, one nit you could address, then you can merge. Approving now for speeeeed

**Comment by dwightguth** [comment]
@traiansf I tested on the haskell integration tests by running scripts/ktest.sh and everything passed.

**Pull request merged**